### DISCUSSION TITLE
File uploading - what type is "photo" property in this example?

### DISCUSSION THREAD

**Discussion by pronode** [livewire/livewire#5361] [1. Help]
Hi, this is an example taken from [https://laravel-livewire.com/docs/2.x/file-uploads](url)
`use Livewire\WithFileUploads;
 
class UploadPhoto extends Component
{
    use WithFileUploads;
 
    public $photo;
 
    public function save()
    {
        $this->validate([
            'photo' => 'image|max:1024', // 1MB Max
        ]);
 
        $this->photo->store('photos');
    }
}`

When I've done` dd($this->photo)` it turns out that it should be `Livewire\TemporaryUploadedFile` but after typing it to `public TemporaryUploadedFile $photo` I get an error `"Cannot assign string to property App\Http\Livewire\MediaUploader::$photo of type Livewire\TemporaryUploadedFile"`.

So am I missing something or I just can't use typed properties with Livewire?

**Comment by joshhanley** [comment]
@pronode a property that is used for a file upload will initially receive a string which Livewire then converts to a `TemporaryUploadedFile` object once it has processed it. So I'm not sure if type hinting will work for a file upload property. But type hints for other properties should work.

Hope this helps!
